### ISSUE TITLE
Measurements sent to InfluxDB seems to be string despite conversion to INTEGER

### ISSUE THREAD

**Issue by mastier** [toni-moreno/snmpcollector#384]
There was some similar issue, but there the field was set as tag, here is different.

![screen shot 2018-11-13 at 18 30 46](https://user-images.githubusercontent.com/1609672/48431670-8ba5ff80-e772-11e8-8609-a19c0306718d.png)

I get counter values from Huawei switches
![screen shot 2018-11-13 at 18 14 50](https://user-images.githubusercontent.com/1609672/48431709-9e203900-e772-11e8-8762-c6436e2804ef.png)

Please include this information:
- What SnmpCollector version are you using? 
snmpcollector v0.8.0 (git: 9e2519a ) built at [2018-11-05 07:11:36]
- What OS are you running snmpcollector on? Ubuntu 14.04 LTS
- What did you do? All set, the measurement are sent to InfluxDB but seems they are strings
```
> show field keys
name: ipIfStats
fieldKey        fieldType
--------        ---------
ifHCInOctets    string
ifHCOutOctets   string
```
- What was the expected result? Field value should be integer
- What happened instead? They are strings

When the conversion set to Float the data is correctly stored in Influxdb (as float of course).
However now I noticed the same like in other bug report that cannot change value conversion to FLOAT. Made an commet in issue #379
Influxdb version 1.1.1

I attached log
[HuaweiCE12800.zip](https://github.com/toni-moreno/snmpcollector/files/2577359/HuaweiCE12800.zip)

**Comment by toni-moreno** [comment]
Hi @mastier, I have solved 2 errors in both the webUI Conversion Selector and also in conversion when metrics are sent as tags. 

Perhaps your issue have to be with se webUI that is showing you conversion = INTEGER ( 1 ) but really in the database you could have configured =STRING ( 3 ) 

Could you please show us the config in the database ? (you can select by ) 
````bash
 sqlite3 snmpcollector.db 
select id,conversion from snmp_metric_cfg where id='ifHCInOctets';
select id,conversion from snmp_metric_cfg where id='ifHCOutOctets';
````

If this is your case, you should only update to the next version with the fix when released or you can yourself compile snmpcollector from sources.

Thank you very much!

**Comment by mastier** [comment]
I've built new one

**Comment by mastier** [comment]
I've managed to built new version! Hooray :P 
```root@node-292:~# snmpcollector --version
snmpcollector v0.8.0 (git: 735ca2b ) built at [2018-11-14 17:08:18]```
Btw. "built at" is misleading, as it is actually commit time.

The latest fixed issue with saving when changing between float and integer.
However when set INTEGER the issue persist. In the database it is 1
```sqlite> select id,conversion from snmp_metric_cfg where id='ifHCInOctets';
ifHCInOctets|1
sqlite> select id,conversion from snmp_metric_cfg where id='ifHCOutOctets';
ifHCOutOctets|1```

I removed the shard and still the same
```> show field keys
name: ipIfStats
fieldKey        fieldType
--------        ---------
ifHCInOctets    string
ifHCOutOctets   string```

I don't know actually how InfluxDB decides of datatype, I've read somewhere that if you have big integer it may treat it as string, until explicitly specify 'i' at the end. As Influx preserves integers as uint64.

**Comment by mastier** [comment]
![image](https://user-images.githubusercontent.com/1609672/49878994-489a8300-fe29-11e8-8d7d-f2b78a92fd67.png)
Values are sent with double quotes when conversion to Integer is set, that might cause Influx to interpret them as string.

**Comment by toni-moreno** [comment]
Hi @mastier  As you discovered influx treat data depending how is sent.

- data=0 (as float)
- data="0" (as string)
- data=0i (as integer 64b)
- data=0u (as unsigned integer 64b)

Anyway I'm worried about what is happening you please config your device in debug mode and let me inspect the logs when you force conversion as integer . Also will be helpful if you can show the complete snmpmetric config in the database

````bash
sqlite> select*  from snmp_metric_cfg where id='ifHCOutOctets';
sqlite>select * from snmp_metric_cfg where id='ifHCInOctets';
````
please attach log files again ( in debug mode)  and I will review ASAP.

**Comment by mastier** [comment]
I enclosed with debug level and snmp debug on

[xx_float.log](https://github.com/toni-moreno/snmpcollector/files/2676336/xx_float.log)
[xx_integer.log](https://github.com/toni-moreno/snmpcollector/files/2676337/xx_integer.log)

In the second, I change to INTEGER only on ifHCOutOctets.

```
sqlite> select*  from snmp_metric_cfg where id='ifHCOutOctets';
ifHCOutOctets|ifHCOutOctets||.1.3.6.1.2.1.31.1.1.1.10|Counter64|0|0.0|0.0|0||1
sqlite> select * from snmp_metric_cfg where id='ifHCInOctets';
ifHCInOctets|ifHCInOctets||.1.3.6.1.2.1.31.1.1.1.6|Counter64|0|0.0|0.0|0||0
```

**Comment by mastier** [comment]
Need anything else or is it just matter of implementation. Need help ?

**Comment by syepes** [comment]
I have also just detected this issue with v0.8.0 and have found the following workaround.
Set the conversion to "FLOAT" even knowing the actual value is not a float, this actually sets value to "0".
Now the values a being sent to InfluxDB as INT's and not strings.

![formselector](https://user-images.githubusercontent.com/333081/51281414-9e42ec00-19e2-11e9-9f27-d8e2b09807a2.jpg)

**Comment by topranks** [comment]
I have noticed the same issue - and used the "float" workaround also.

**Comment by toni-moreno** [comment]
Hi @syepes @topranks thank you for working in this workaround, we will work in the fix  ASAP .

**Comment by topranks** [comment]
Toni thanks for the reply.  I can do some tests if you would like.

I haven't 100% validated this but the pattern that occurs for me is usually like this:

- Add a metric as type 'COUNTERXX' (my script defaults to adding counters as this type)
- Realise I need the actual counter values, so change from COUNTERXX to Counter32 or Counter64 manually using GUI.

After changing to Counter32 or Counter64 the type I see in GUI is 'INTEGER' but a string gets written to Influx.  Setting to float seems to be a workaround as I said.

Let me know if you want I can do a test run of the above and capture the interaction between GUI and back end.

**Comment by mnantel** [comment]
influxpoint.go, function getInfluxPoint - issue appears to be when creating client.Point.
I added code in that function to generate before and after output:

			fmt.Println("## 1/2 ## Fields #####")
			b, err := json.MarshalIndent(Fields, "", "  ")
			if err != nil {
				fmt.Println("error:", err)
			}
			fmt.Print(string(b))
			// here we can chek Fields names prior to send data
			m.Debugf("FIELDS:%+v TAGS:%+v", Fields, Tags)
			pt, err := client.NewPoint(m.cfg.Name, Tags, Fields, t)
			if err != nil {
				m.Warnf("error in influx point creation :%s", err)
				measError++
			} else {
				m.Debugf("GENERATED INFLUX POINT[%s] index [%s]: %+v", m.cfg.Name, idx, pt)
				ptarray = append(ptarray, pt)
				measSent++
				vIdx.Valid = true
			}
			fmt.Println("## 2/2 ## client.Point #####")
			fmt.Println(pt)

This yields samples such as:
## 1/2 ## Fields #####
{
  "ifInOctets": 4202926039
}## 2/2 ## client.Point #####
testmeasure,device=fsw524,ifIndex=port4(primary) ifInOctets="4202926039" 1549485960002925000
## 1/2 ## Fields #####
{
  "ifInOctets": 0
}## 2/2 ## client.Point #####
testmeasure,device=fsw524,ifIndex=port3(primary) ifInOctets="0" 1549485960002925000
## 1/2 ## Fields #####
{
  "ifInOctets": 0
}## 2/2 ## client.Point #####
testmeasure,device=fsw524,ifIndex=port12(primary) ifInOctets="0" 1549485960002925000


client.NewPoint appears to take a perfectly valid integer and output a string in the Point. Not sure why..

**Comment by topranks** [comment]
That's interesting....   so the issue is with the InfluxDB go client bundled?

I wonder if a trailing 'i' is needed after the number?  The InfluxDB page says integers should be _Signed 64-bit integers (-9223372036854775808 to 9223372036854775807). Specify an integer with a trailing i on the number. Example: 1i._

I've installed with binary here but if I get some time I'll try to get it working from source and see if changing this has any effect.

There was another issue similar with Influx some time back may or may not be relevant:

https://github.com/intelsdi-x/snap-plugin-publisher-influxdb/issues/136

**Comment by CommanderKagome** [comment]
I have the same issue.
snmpcollector 0.8 (debian package)
influxDB 1.7.6

What I notice is that some Counter32 Values are > 2^31 on my snmp querys.
That float workaround will work - but the database type is then float ..

Is there any date when this will be fixed?

**Comment by hoosee** [comment]
I also have the same problem:

SNMPCollector 0.8
InfluxDB 1.7.9

When I let snmpcollector create the measurements, this happens:

```
> show field keys
name: interface
fieldKey      fieldType
--------      ---------
ifHCInOctets  string
ifHCOutOctets float

```

```
> select * from interface
name: interface
time                device    ifHCInOctets  ifHCOutOctets ifName
----                ------    ------------  ------------- ------
1573561860000000000 MYLAB-FW1 0             299492222     port1
1573561860000000000 MYLAB-FW1 0             0             port2
1573561860000000000 MYLAB-FW1 0             23700         port3
1573561860000000000 MYLAB-FW1 0             2652          port4

```

When I created measurement by hand:

```
> insert test1,device=testi,ifName=testi ifHCInOctets=5,ifHCOutOctets=5
> insert test2,device=testi,ifName=testi ifHCInOctets="5",ifHCOutOctets=5
> insert test3,device=testi,ifName=testi ifHCInOctets="5i",ifHCOutOctets=5i
> show field keys
name: test1
fieldKey      fieldType
--------      ---------
ifHCInOctets  float
ifHCOutOctets float

name: test2
fieldKey      fieldType
--------      ---------
ifHCInOctets  string
ifHCOutOctets float

name: test3
fieldKey      fieldType
--------      ---------
ifHCInOctets  string
ifHCOutOctets integer

```

**Comment by hoosee** [comment]
Also confirming what mastier already posted:

```
sqlite> select id,conversion from snmp_metric_cfg
FORTINET-FORTIGATE-MIB::fgSysCpuUsage|1
FORTINET-FORTIGATE-MIB::fgSysMemUsage|1
FORTINET-FORTIGATE-MIB::fgSysSesCount|1
FORTINET-FORTIGATE-MIB::fgSysDiskUsagePrc|0
```

```
Accept-Encoding: gzip fgSys,device=MYLAB-FW1 fgSysCpuUsage="9",fgSysDiskUsagePrc=1.3936934318230736,fgSysMemUsage="45",fgSysSesCount="2197" 1573577520
```

**Comment by toni-moreno** [comment]
Hello @syepes , @hoosee , @mnantel  ,  @topranks . New released 0.10 version , works from  a new influxdb 1.X Client from  https://github.com/toni-moreno/snmpcollector/commit/b9733ccb20ec2699e12dc679f85503abd1fac5d9 . This issue should not happen anymore. Feel free to reopen if needed.
